src/test/ui/const-generics/const_evaluatable_checked/abstract-const-as-cast-2.rs

```diff
@@ -0,0 +1,13 @@
+#![feature(const_evaluatable_checked, const_generics)]
+#![allow(incomplete_features)]
+
+trait Evaluatable<const N: u128> {}
+impl<const N: u128> Evaluatable<N> for () {}
+
+struct Foo<const N: u8>([u8; N as usize])
+//~^ Error: unconstrained generic constant
+//~| help: try adding a `where` bound using this expression: `where [(); N as usize]:`
+where
+    (): Evaluatable<{N as u128}>;
+
+fn main() {}
```

Comment:
Please add some more tests. Like doing `N as usize as usize` is not implied by a `N as usize` bound

Reply:
oh yeah, thats good thinking thanks!